pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go

```diff
@@ -122,10 +129,24 @@ func (c *cyclonedxParser) addPackages(cdxBom *cdx.BOM) {
 			CPEs:     []string{comp.CPE},
 			NodeData: *assembler.NewObjectMetadata(c.doc.SourceInformation),
 		}
-		c.rootPackage.depPackages = append(c.rootPackage.depPackages, parentPackages{
+		parentPkg := parentPackages{
 			curPackage:  curPkg,
-			depPackages: []parentPackages{},
-		})
+			depPackages: []*parentPackages{},
+		}
+		c.rootPackage.depPackages = append(c.rootPackage.depPackages, &parentPkg)
+		pkgMap[comp.BOMRef] = &parentPkg
+	}
+
+	if cdxBom.Dependencies != nil {
+		for _, deps := range *cdxBom.Dependencies {
+			if topPkg, found := pkgMap[deps.Ref]; found {
+				for _, depPkg := range *deps.Dependencies {
+					if depPkg, exist := pkgMap[depPkg]; exist {
+						topPkg.depPackages = append(topPkg.depPackages, depPkg)
```

Comment:
yea it's a little difficult to reason about thinking about it as a "top" package, probably needs some renaming..  This loop is pretty nested as well, can refactor this or make some of these continues so it doesn't next as much?

Reply:
done!